### DISCUSSION TITLE
Advise needed with digital POS receipt

### DISCUSSION THREAD

**Discussion by vb90vb90** [microsoft/Dynamics365Commerce.Solutions#166] [Q&A] [answered]
We have requirement to do customization for our customer on the sales order bill when we have a transaction in POS. The sales order bill that it is printed now has the full message body that Is sent from POS.
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/129947782/0c6e7552-60e9-4b77-9a68-8997fd5450c5)

Example in the CONTOSO system is the one outlined in the screen below.

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/129947782/9455aa9b-8798-4e2d-987c-720904730709)

From what we have noticed this is the only mail that is sent directly and synchronously from POS when the sales order Bill is created.

In this email template there are only two places’ holders %message% and %receiptid%
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/129947782/401e843f-19a0-4b57-b255-2da1368fc679)

When I have tried to debug, I see the full message is sent from POS.
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/129947782/633e8734-e404-4c0a-ad55-c8cff6d27f01)

Our client wants to modify the Sales receipt bill to have the usual place holders like the rest of the Email notification types.
I see that these values for the other email notification types are filled out from the sale order data in in F&O.
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/129947782/a1629923-fbe4-4231-9d11-c4f181162050)

Can you please give us an advice what is best solution to implement this customer requirement?

**Comment by madyke** [comment]
Hello @vb90vb90 - if I'm understanding your scenario correctly, you shouldn't need to write .xpp code to modify the sales order receipt. You should be able to do this either directly in the receipt designer or by creating an extension in Store Commerce to control what content and placeholders are sent for this receipt. These docs should have more info that you can use to get started:
[Extend Commerce Store receipts](https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/retail-sdk/retail-sdk-samples)
[Send email receipts from Store Commerce](https://learn.microsoft.com/en-us/dynamics365/commerce/email-receipts)

**Reply by vb90vb90** [reply]
Hi Matt Dyke, I am not sure if you looked at the document but Basically what we need is to update the Design of the Email that sends the Digital Receipt - in other words, we want to have design options for **e-mail templates for the receipt**, instead of just %message% placeholder.  
Please see the attached images of what we mean

This is the required design
![MicrosoftTeams-image (1)](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/129947782/493788f2-3d0e-4fac-8117-5cd9561c5226)  is what we hope to achieve, in terms of design capabilities (fonts, colors, size, spacing). 

but the standard is a design like this
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/129947782/db1ae306-6e2b-4feb-82c3-d0beb1da594b) where "%message% Design" is used, which is just a placeholder for the POS Receipt (Receipt Designer), which does not allow us to customize the style/design in any way. Receipt Designer is not an option, as it mostly allows the positioning of fields.

**Reply by madyke** [reply, marked as answer]
Hello @vb90vb90, thanks for the clarification. I discussed this with our receipt expert today but unfortunately he informed me that there isn't a way for customers to do this currently. There are no extension points customers can take advantage of to do this other than taking the whole %message% placeholder.

Our team has discussed in the past breaking down the %message% placeholder into more granular placeholders, but we haven't had the time to do it. If the customer needs this as a requirement the best they can do for now is to submit it as a suggestion through the Ideas portal.
